azext_edge/edge/providers/support/base.py

```diff
@@ -24,7 +24,15 @@
 DAY_IN_SECONDS: int = 60 * 60 * 24
 
 
-def process_crd(group: str, version: str, kind: str, plural: str, api_moniker: str, file_prefix: Optional[str] = None):
+def process_crd(
+    group: str,
+    version: str,
+    kind: str,
+    plural: str,
+    api_moniker: str,
+    child_group: str = "",
```

Comment:
I also like Optional[str] = None and prefer name change

Reply:
I'm kind of struggled with using None or "" for this child group, if we use None as default, isn't that we still need to add check `if not child_group: child_group=""`? so why not default it to ""?😥

Reply:
The spirit of the comment is around refactoring how the zip file path is constructed instead of today using a format string with static placement of the nested directory segment (relying on "" as default) and the caller having to include a trailing slash.